amun/core.py

```diff
@@ -66,11 +66,23 @@ def create_inspect_buildconfig(openshift: OpenShift, inspection_id: str, dockerf
     _LOGGER.debug("OpenShift response for getting Amun inspect BuildConfig template: %r", response)
 
     template = response['items'][0]
-    openshift.set_template_parameters(
-        template,
-        AMUN_INSPECTION_ID=inspection_id,
-        AMUN_GENERATED_DOCKERFILE=dockerfile
-    )
+    if 'build' in specification:
+        build_specification = specification['build']['response']
+        BUILD_MEMORY = build_specification['memory']
+        BUILD_CPU = build_specification['cpu']
+        openshift.set_template_parameters(
+            template,
+            AMUN_INSPECTION_ID=inspection_id,
+            AMUN_GENERATED_DOCKERFILE=dockerfile,
+            AMUN_BUILD_CPU=BUILD_CPU,
+            AMUN_BUILD_MEMORY=BUILD_MEMORY,
```

Comment:
You provided defaults in the template. A user can submit just cpu or memory requirements (based on the comment above) so somethink like this can be more clear:

```python
parameters = {
  'AMUN_INSPECTION_ID': inspection_id,
  'AMUN_GENERATED_DOCKERFILE': dockerfile
}

if 'build' in specification:
   if 'cpu' in specification['build']:
     parameters['AMUN_BUILD_CPU'] = specification['build']['cpu']
   if 'memory' in specification['build']:
     parameters['AMUN_BUILD_CPU'] = ...

openshift.set_template_parameters(template, **parameters)
```

Here and in the build bellow.